tsfc/fem.py

```diff
@@ -694,7 +694,13 @@ def take_singleton(xs):
 
 
 def compile_ufl(expression, interior_facet=False, point_sum=False, **kwargs):
-    context = PointSetContext(**kwargs)
+
+    # kwargs must be context kwargs for a PointSetContext or GemPointContext
+    point_expr = kwargs.get("point_expr")
+    if point_expr:
+        context = GemPointContext(**kwargs)
+    else:
+        context = PointSetContext(**kwargs)
```

Comment:
I suspect it makes more sense to refactor and make the caller responsible for constructing the context. That way we're not reliant on `point_expr` never being a kwarg for the `PointSetContext` case.

So:

```
def compile_ufl(expression, context, interior_facet=False, point_sum=False):
   """Translate a UFL expression to GEM.

   :arg expression: ...
   :arg context: translation context.
   ...
   """
   ...
```